internal/engine/crd_test.go

```diff
@@ -0,0 +1,271 @@
+// exist at runtime in the graph; do not look for it with e.g. LookupPath().
+var goldenBucketFirstSchema = `import "strings"
+
+_#def
+_#def: {
+	// APIVersion defines the versioned schema of this representation
+	// of an object. Servers should convert recognized schemas to the
+	// latest internal value, and may reject unrecognized values.
+	// More info:
+	// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+	apiVersion: "source.toolkit.fluxcd.io/v1beta1"
+
+	// Kind is a string value representing the REST resource this
+	// object represents. Servers may infer this from the endpoint
+	// the client submits requests to. Cannot be updated. In
+	// CamelCase. More info:
+	// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	kind: "Bucket"
+	metadata: {
+		name:      string
+		namespace: string
+		labels?: {
+			[string]: string
+		}
+		annotations?: {
+			[string]: string
+		}
+	}
+
+	// BucketSpec defines the desired state of an S3 compatible bucket
+	spec?: {
+		// AccessFrom defines an Access Control List for allowing
+		// cross-namespace references to this object.
+		accessFrom?: {
+			// NamespaceSelectors is the list of namespace selectors to which
+			// this ACL applies. Items in this list are evaluated using a
+			// logical OR operation.
+			namespaceSelectors: [...{
+				// MatchLabels is a map of {key,value} pairs. A single {key,value}
+				// in the matchLabels map is equivalent to an element of
+				// matchExpressions, whose key field is "key", the operator is
+				// "In", and the values array contains only "value". The
+				// requirements are ANDed.
+				matchLabels?: {
+					[string]: string
+				}
+				...
+			}]
+			...
+		}
+
+		// The bucket name.
+		bucketName: string
+
+		// The bucket endpoint address.
+		endpoint: string
+
+		// Ignore overrides the set of excluded patterns in the
+		// .sourceignore format (which is the same as .gitignore). If not
+		// provided, a default will be used, consult the documentation
+		// for your version to find out what those are.
+		ignore?: string
+
+		// Insecure allows connecting to a non-TLS S3 HTTP endpoint.
+		insecure?: bool
+
+		// The interval at which to check for bucket updates.
+		interval: string
+
+		// The S3 compatible storage provider name, default ('generic').
+		provider?: "generic" | "aws" | "gcp" | *"generic"
+
+		// The bucket region.
+		region?: string
+
+		// The name of the secret containing authentication credentials
+		// for the Bucket.
+		secretRef?: {
+			// Name of the referent.
+			name: string
+			...
+		}
+
+		// This flag tells the controller to suspend the reconciliation of
+		// this source.
+		suspend?: bool
+
+		// The timeout for download operations, defaults to 60s.
+		timeout?: string | *"60s"
+		...
+	}
+
+	// BucketStatus defines the observed state of a bucket
+	status?: {
```

Comment:
I’m not sure if `status`, which is a read-only field, should be exported. At apply time, Timoni removes the `status`, even if it’s specified.

Reply:
Yeah, in other discussions i've had, when to include particular subelements in schemas has often been quite murky. Generally depends on the user/use case for that variant of the schema.

i'll pull it out of the overall resource def, but retain the standalone definition.